src/keys.rs

```diff
@@ -398,6 +478,14 @@ impl From<&FullViewingKey> for IncomingViewingKey {
 }
 
 impl IncomingViewingKey {
+    /// Serializes an Orchard incoming viewing key to its raw encoding
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut result = [0u8; 64];
+        result.copy_from_slice(self.dk.to_bytes());
+        result[32..].copy_from_slice(&self.ivk.0.to_bytes());
+        result
+    }
+
     /// Parses an Orchard incoming viewing key from its raw encoding.
     pub fn from_bytes(bytes: &[u8; 64]) -> CtOption<Self> {
         NonZeroPallasBase::from_bytes(bytes[32..].try_into().unwrap()).map(|ivk| {
```

Comment:
Just below here, we fail to check that ivk is in the range {1..q<sub>P</sub>-1}:

![image](https://user-images.githubusercontent.com/643204/128520491-ff95d0ca-f4e9-4174-99f9-a4dc56ef9f92.png)

Reply:
I think `NonZeroPallasBase::from_bytes()` checks this.